lib/src/tar/tar_file.dart

```diff
@@ -98,10 +97,9 @@ class TarFile {
     }
 
     if (isFile && fileSize > 0) {
-      final remainder = fileSize % 512;
-      var skiplen = 0;
+      final remainder = fileSize & 0x1ff;
       if (remainder != 0) {
-        skiplen = 512 - remainder;
+        final skiplen = 0x200 ^ remainder;
```

Comment:
I don't think this is a high frequency part of the code. In that case, the slower but clearer operators are better than tricky bitwise operators. I'll never remember that `0x200 ^ remainder` is the same as `512 - remainder` 6 months from now.

Reply:
I agree, but what about using such code along with some comment ? It is not a high frequency part of the code but you have to think that your library (package) is used by thousands of user, many times a day, so even a small improvement can overall have a very positive global output. I agree that for personal projects its rather pointless, but I think libraries should always be optimized as much as possible, without hurting too much readability. But I do understand if you do not want these changes as you are the project maintainer it is up to you!

Reply:
Adding a comment describing the expected behavior of the bit logic will be good. Also need to verify that it does what it's expected to do in Javascript since that sometimes has some different behavior for bit logic (at least with bit shifts).